smart-send-logistics/includes/frontend/class-ss-shipping-frontend.php

```diff
@@ -63,56 +74,113 @@ public function display_ss_pickup_points($method, $index)
 
-                    if (!empty($ss_agents)) {
+                    $this->carries = $carrier;
+                    $restapiend = '';
+                    $ss_agents = $this->find_closest_agents_by_address($carrier, $country, $postal_code, $city, $street, $restapiend);
 
+                    if (!empty($ss_agents)) {
                         $ss_setting = SS_SHIPPING_WC()->get_ss_shipping_settings();
 
                         $ss_agent_options = array();
                         if (!isset($ss_setting['default_select_agent']) || $ss_setting['default_select_agent'] == 'no') {
-                            $ss_agent_options[0] = __('- Select Pick-up Point -',
-                                    'smart-send-logistics');
+                            $ss_agent_options[0] = __('- Select Pick-up Point -', 'smart-send-logistics');
                         }
 
                         foreach ($ss_agents as $key => $agent) {
                             $formatted_address = $this->get_formatted_address($agent);
-                            $ss_agent_options[ $agent->agent_no ] = $formatted_address;
+                            $ss_agent_options[$agent->agent_no] = $formatted_address;
                         }
 
-                        woocommerce_form_field( 'ss_shipping_store_pickup', array(
-                            'type'          => 'select',
-                            'options'       => $ss_agent_options,
-                            'input_class'   => array('ss-agent-list'),
-                            )
-                        );
-
+                        woocommerce_form_field('ss_shipping_store_pickup', array(
+                            'type' => 'select',
+                            'options' => $ss_agent_options,
+                            'input_class' => array('ss-agent-list'),
+                        ));
                     } else {
-                        echo '<div class="woocommerce-info ss-agent-info">' . __('Shipping to closest pick-up point',
-                                'smart-send-logistics') . '</div>';
+                        echo '<div class="woocommerce-info ss-agent-info">' . __('Shipping to closest pick-up point', 'smart-send-logistics') . '</div>';
                     }
                 } else {
-                    echo '<div class="woocommerce-info ss-agent-info">' . __('Enter shipping information',
-                            'smart-send-logistics') . '</div>';
+                    echo '<div class="woocommerce-info ss-agent-info">' . __('Enter shipping information', 'smart-send-logistics') . '</div>';
                 }
             }
         }
 
-	    /**
-	     * Find the closest agents by address
+        /**
+         * wordpress endpointn call registeration call back function
+         * @ $request
+         * 
+         * **/
+
+        function get_pickup_points_callback(WP_REST_Request $request)
```

Comment:
Private or public?

This should be moved to a dedicated class.

Reply:
moved to a dedicated class in the fronted folder as a public function